webapp/src/selectors.js

```diff
@@ -46,3 +46,24 @@ export const getSiteURL = () => {
 
     return siteURL;
 };
+
+const getCurrentlyOpenRhsPluginId = (state) => {
+    const rhs = state.views.rhs;
+    if (rhs.rhsState !== 'plugin') {
+        return '';
+    }
+    const available = state.plugins.components.RightHandSidebarComponent || [];
+    if (!available.length) {
+        return '';
+    }
```

Comment:
This is an undocumented way to find out what plugin is showing in the RHS, and may change in the future. I suggest adding something like this to the [RHS component](https://github.com/mattermost/mattermost-plugin-todo/blob/43c3c30c6b587f4d9ff6ae61a6275704ca45c811/webapp/src/components/sidebar_right/sidebar_right.jsx#L82) in this plugin:

```js
componentDidMount() {
    this.props.setVisible(true);
}

componentWillUnmount() {
    this.props.setVisible(false);
}
```

Then the `setVisible` function would update something in the redux store that represents if this plugin's RHS component is currently open.

Reply:
Awesome! I incorporated your suggested approach in the new commit. Thanks! 💯